### ISSUE TITLE
artichoke-frontend depends on mruby backend internals

### ISSUE THREAD

**Issue by lopopolo** [artichoke/artichoke#234]
`airb` depends on:

- parser
- context

`string_scan_bench` depends on:

- global variable get

Add APIs to artichoke-core for these items.

**Comment by lopopolo** [comment]
Blocked on GH-116.

```
error[E0624]: method `inner` is private
  --> artichoke-frontend/src/bin/string_scan_bench.rs:45:83
   |
45 |     unsafe { sys::mrb_gv_set(mrb, interp.0.borrow_mut().sym_intern("$data"), data.inner()) }
   |
```

**Comment by lopopolo** [comment]
`airb`'s dependency on parser internals was addressed a long time ago:

- https://github.com/artichoke/artichoke/pull/467

`string_scan_bench` was also removed a long time ago:

- https://github.com/artichoke/artichoke/pull/362

This ticket will be closed out in spirit once #1760 lands.
